### PR TITLE
Fix key bindings in vi mode and add compatibility for Fisher 4.2.0+

### LINKED ISSUE #4: Doesn't work with fish_vi_key_bindings
Substitution doesn't occur if using vi keybindings rather than default.

### PR DESCRIPTION
**Pull request by akuntsch**
Fixes #4, #7.

I also added a Fisher uninstall function which erases the key bindings on removal via Fisher.

### CODE REVIEW

**Comment by nzig** [comment]
This would also fix issues with Fisher: https://github.com/jorgebucaran/fisher/issues/651

File: conf.d/plugin-bang-bang.fish [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+if test "$fish_key_bindings" = 'fish_vi_key_bindings'
+    bind --mode insert ! __history_previous_command
+    bind --mode insert '$' __history_previous_command_arguments
+end
+
+function _plugin-bang-bang_uninstall --on-event plugin-bang-bang_uninstall
+    bind --erase --all !
+    bind --erase --all '$'
+    echo "plugin-bang-bang key bindings removed"
+
```
Comment (scorphus) [inline conf.d/plugin-bang-bang.fish]: How about removing this line? Implementation details are implementation details :-)
```suggestion
```

**Review by scorphus — CHANGES_REQUESTED** [review]
Thanks for contributing! This is not yet fullt supported by OMF, but as there's no `uninstall` hook, I guess there won't be real difference for current users.

Please mind my change request.

**Comment by akuntsch** [comment]
I've removed the `echo` and added an `uninstall` hook for OMF which removes the keybindings.

**Review by scorphus — APPROVED** [review]

**Comment by scorphus** [comment]
Thanks @akuntsch !

**Pull request merged**